src/test/parseRuntimeAndValidateSDK.spec.ts

```diff
@@ -0,0 +1,81 @@
+import * as sinon from "sinon";
+import { expect } from "chai";
+import * as utils from "../utils";
+import * as runtime from "../parseRuntimeAndValidateSDK";
+import * as checkFirebaseSDKVersion from "../checkFirebaseSDKVersion";
+import { FirebaseError } from "../error";
+// have to require this because no @types/cjson available
+// tslint:disable-next-line
+const cjson = require("cjson");
+
+describe("getRuntimeName", () => {
+  it("should properly convert raw runtime to human friendly runtime", () => {
+    expect(runtime.getHumanFriendlyRuntimeName("nodejs6")).to.contain("Node.js");
+  });
+});
+
+describe("getRuntimeChoice", () => {
+  const sandbox = sinon.createSandbox();
+  let cjsonStub: sinon.SinonStub;
+  let warningSpy: sinon.SinonSpy;
+  let SDKVersionStub: sinon.SinonStub;
+
+  beforeEach(() => {
+    cjsonStub = sandbox.stub(cjson, "load");
+    warningSpy = sandbox.spy(utils, "logWarning");
+    SDKVersionStub = sandbox.stub(checkFirebaseSDKVersion, "getFunctionsSDKVersion");
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it("should return node 6 if package.json engines field is set to node 6 and print warning", () => {
+    cjsonStub.returns({ engines: { node: "6" } });
+    SDKVersionStub.returns("2.0.0");
+
+    expect(runtime.getRuntimeChoice("path/to/source")).to.eventually.deep.equal("nodejs6");
```

Comment:
For tests that are async (`eventually` is the part that's async here), mocha expects a promise to be returned. The test function needs to be `async` and it either has to return a single `expect...eventually`, or do `await expect...eventually`

(the failure modes for this are weird because it will look like unhandled promise rejections if this test fails)

Reply:
this method is no longer async, so this no longer applies.